git_machete/__init__.py

```diff
@@ -1 +1 @@
-__version__ = '3.11.1'
+__version__ = '3.11.2'
```

Comment:
Btw! this PR isn't correctly names, `get_inferred_remote_for_fetching_of_branch` still doesn't (and shouldn't!) take into account the branch tracking data (which is typically understood as `branch.{...}.remote` git config)

Reply:
It's the `get_strict_...` versions that rely on `branch.{...}.remote` git config

Reply:
You may actually add some docstrings about that in code to make it clearer for the future devs